packages/channels/src/base/service.ts

```diff
@@ -38,15 +39,20 @@ export abstract class ChannelService<
   async setup() {}
 
   async start(scope: string, config: TConfig) {
-    const state = await this.create(scope, config)
+    await this.addState(scope, config)
+    await this.emit('start', { scope })
+  }
 
-    if (this.manager) {
-      this.manager.set(scope, state)
-    } else {
-      this.states[scope] = state
+  async test(scope: string, config: TConfig) {
+    try {
+      await this.addState(scope, config)
+      await this.emit('test', { scope })
+      return true
```

Comment:
Yea whatever it's kinda weird since you never return false. It would make more sense to return an error object in this case, but it would make the look worse so I thought it was a compromise

Reply:
Maybe we could have a try catch at the root of the test function and then returned whatever is catched there. That would be less weird

Reply:
Or you could just throw an error or return void? I feel like returning true is unnecessary. But, on the other hand, it's a minor personal preference, so it doesn't really matter!